### PR TITLE
Fixed config issue with AWS SDK v3 in AwsS3ResolverFactory

### PR DESCRIPTION
**Pull request by kieste**
I just modified the pull request from glenn-- so that his changes pass PHP 5.3 checks.

### CODE REVIEW

**Comment by makasim** [comment]
What if we add a new factory for v3 sdk? Same time we can deprecate curent aws factory. 

It would be good if you can add some docs on when to use what factory.

Sounds good?

**Comment by makasim** [comment]
@kieste are you planning to finish this PR?

**Comment by kieste** [comment]
Yes I will fix this before end of the year.

**Comment by kieste** [comment]
@makasim separated old and new SDK client config code into different factory classes. Added a note to the documentation.

File: LiipImagineBundle.php [hunk 1]
```diff
@@ -32,7 +33,11 @@ public function build(ContainerBuilder $container)
         $extension = $container->getExtension('liip_imagine');
 
         $extension->addResolverFactory(new WebPathResolverFactory());
-        $extension->addResolverFactory(new AwsS3ResolverFactory());
+        if (defined('\Aws\Sdk::VERSION') && version_compare(\Aws\Sdk::VERSION, '3.0.0', '>=')) {
```
Comment (makasim) [inline LiipImagineBundle.php]: register both, and let a developer decided which one to use. Also would be good to add an exception if a resolver is used with wrong sdk.

File: Resources/doc/cache-resolver/aws_s3.rst [hunk 2]
```diff
@@ -42,6 +42,18 @@ Create resolver using factory
                   put_options:
                       CacheControl: 'max-age=86400'
 
+If you use `aws-sdk-php`_ library version >= 3.0.0 client config credentials
+must be an associative array containing key and secret.
+
+.. code-block:: yaml
+
+    aws_s3:
```
Comment (makasim) [inline Resources/doc/cache-resolver/aws_s3.rst]: I meant the resolver for sdk v3 has to have separate name, like aws_s3_sdkv3

**Comment by kieste** [comment]
@makasim after taking a closer look at my current setup I realized that I am not using the factories any more. (Currently I am creating the AwsS3Resolver as a service and injecting it directly into a proxy resolver.)

So I can't finish this pull request now with your suggested changes. (I must progress with my current project.) But I aggree to all of your comments. This totally makes sense. But it's too much effort for me at the moment.

**Comment by makasim** [comment]
Thanks for your contribution, I took care of your PR. Soon in the master

**Pull request closed without merging**